Social: Improve the way Jetpack Social products are displayed.

spsiddarthan opened the pull request:
The Jetpack Social products are shown using its own component, which we wrote just to render social products. After receiving feedback from multiple people, we have decided to display the social products in the same way as we display the backup products. 

<!--
Link a related issue to this PR. If the PR does not immediately resolve the issue,
for example, it requires a separate deployment to production, avoid
using the "fixes" keyword and instead attach the [Status] Fix Inbound label to
the linked issue.
-->

## Proposed Changes

* Show the Jetpack Social plans using radio buttons
* If your existing social subscription is a monthly one, currently on prod, we just take you to the calypso page after clicking on "Manage Subscription". In this PR, we open the modal and display the monthly variants of the prices. The behaviour is unchanged if your existing social subscription is a yearly one. 
* I have also added the changes to include the introductory cost calculation for the monthly plan

## Testing Instructions

<!--
Add as many details as possible to help others reproduce the issue and test the fix.
"Before / After" screenshots can also be very helpful when the change is visual.
-->

Testing the pricing page for siteless checkouts
- Go to http://jetpack.cloud.localhost:3000/pricing and make sure you're able to checkout the Jetpack Social Basic and Advanced plans. 
<img width="1142" alt="Screenshot 2023-02-08 at 18 35 25" src="https://user-images.githubusercontent.com/6594561/217538352-be7b90ae-43bd-40b0-a174-5e3db6757bb6.png">
-  Go to http://jetpack.cloud.localhost:3000/pricing and make sure you're able to checkout the VaultPress backup plan. This step is to make sure that the  VaultPress backup plan, the only other product with an introductory offer gets checked out correctly. 
- Go to http://jetpack.cloud.localhost:3000/pricing and make sure you're able to checkout the Jetpack Boost plan. This step is to make sure that the non-introductory offers plans are being checked out correctly. 


Testing the pricing page for sites with existing social plans, please boot 4 new sites as described below and don't use an existing site since free trials won't be available to the site that has already used them. 

- Boot up a new jetpack site and buy the Jetpack Social Basic monthly plan.
- Boot up a new jetpack site and buy the Jetpack Social Basic yearly plan.
- Boot up a new jetpack site and buy the Jetpack Social Advanced monthly plan.
- Boot up a new jetpack site and buy the Jetpack Social Advanced yearly plan.
- For each of the above sites, Go to http://jetpack.cloud.localhost:3000/pricing/your-jeptpack.site.domain?site=your-jeptpack.site.domain and click on the manage subscription button next to "Social". 
<img width="688" alt="Screenshot 2023-02-08 at 18 43 53" src="https://user-images.githubusercontent.com/6594561/217540075-a075cadb-a763-47e8-a8d1-44b08962a175.png">

- For the product you already own, you will be able to see the "Manage Subscription" button and for the product you don't own, you will see the add to cart button. 
<img width="561" alt="Screenshot 2023-02-08 at 18 47 04" src="https://user-images.githubusercontent.com/6594561/217540944-b1da00e5-76ea-4d99-93ff-16576b6d940f.png">



## Pre-merge Checklist

<!--
Complete applicable items on this checklist **before** merging into trunk. Inapplicable items can be left unchecked.

Both the PR author and reviewer are responsible for ensuring the checklist is completed.
-->

- [ ] Has the general commit checklist been followed? (PCYsg-hS-p2)
- [ ] [Have you written new tests](https://wpcalypso.wordpress.com/devdocs/docs/testing/index.md) for your changes?
- [ ] Have you tested the feature in Simple (P9HQHe-k8-p2), Atomic (P9HQHe-jW-p2), and self-hosted Jetpack sites (PCYsg-g6b-p2)?
- [ ] Have you checked for TypeScript, React or other console errors?
- [ ] Have you used memoizing on expensive computations? More info in [Memoizing with create-selector](https://github.com/Automattic/wp-calypso/blob/trunk/packages/state-utils/src/create-selector/README.md) and [Using memoizing selectors](https://react-redux.js.org/api/hooks#using-memoizing-selectors) and [Our Approach to Data](https://github.com/Automattic/wp-calypso/blob/trunk/docs/our-approach-to-data.md)
- [ ] Have we added the "[Status] String Freeze" label as soon as any new strings were ready for translation (p4TIVU-5Jq-p2)?
- [ ] For changes affecting Jetpack: Have we added the "[Status] Needs Privacy Updates" label if this pull request changes what data or activity we track or use (p4TIVU-ajp-p2)?

spsiddarthan committed:
Show Jetpack Social products as a radio button.

spsiddarthan committed:
Added monthly pricing.

spsiddarthan committed:
Merge branch 'trunk' of https://github.com/Automattic/wp-calypso into add/fix-social-product-name

spsiddarthan committed:
Add introductory offer intro for monthly product.

spsiddarthan committed:
Show the modal for the monthly price as well.

spsiddarthan committed:
Fix the introductory offer code for monthly products.

spsiddarthan commented on client/my-sites/plans/jetpack-plans/product-lightbox/constants.ts:
```diff
@@ -35,5 +43,7 @@ export const PRODUCT_OPTIONS_HEADER: Record< string, string > = {
 	[ PRODUCT_JETPACK_SOCIAL_BASIC ]: translate( 'Select your plan:' ),
+	[ PRODUCT_JETPACK_SOCIAL_BASIC_MONTHLY ]: translate( 'Select your plan:' ),
 	[ PRODUCT_JETPACK_SOCIAL_ADVANCED ]: translate( 'Select your plan:' ),
+	[ PRODUCT_JETPACK_SOCIAL_ADVANCED_MONTHLY ]: translate( 'Select your plan:' ),
```
We need the monthly product additions to display the monthly variants for the Jetpack Social product

pablinos replied:
This repetition seems a bit odd to me, but it would need a larger refactor to change it.

spsiddarthan replied:
I agree it can be better, but yeah perhaps a part of a bigger refactor

spsiddarthan commented on client/my-sites/plans/jetpack-plans/product-lightbox/index.tsx:
```diff
@@ -25,7 +27,6 @@ import { Icons } from './icons/icons';
 import { Tags } from './icons/tags';
 import PaymentPlan from './payment-plan';
 import ProductDetails from './product-details';
-import ProductWithVariants from './product-with-variants';
```
This is the custom component we were using to display the jetpack social products, we are getting rid of it now and use the same component that the backup product uses.

pablinos replied:
Ah, that's another reason to use the pattern we already of have

spsiddarthan commented on client/my-sites/plans/jetpack-plans/use-item-price.ts:
```diff
@@ -161,25 +161,23 @@ const useItemPrice = (
-				if ( discountedPrice ) {
-					discountedPriceDuration = 1;
-				}
+		originalPrice = item.term !== TERM_MONTHLY ? getMonthlyPrice( itemCost ) : originalPrice;
```
The introductory offer isn't being displayed correctly for the monthly products, this fixes that. I am aware there is chat around removing the trial for the monthly product, but the jetpack social product is going to retain the monthly variant, per the latest information I know.

CodeyGuyDylan replied:
FYI they might be removed for Social too: pbNhbs-5iq-p2#comment-13309

pablinos replied:
We will still need this after removing the monthly trial? This change seems to do a whole lot more for monthly plans, that wouldn't have happened before. We might want to check it doesn't affect the other products.

spsiddarthan commented on packages/calypso-products/src/translations.tsx:
```diff
@@ -158,8 +158,8 @@ export const getJetpackProductsDisplayNames = (): Record< string, TranslateResul
-	const socialBasic = translate( 'Social Basic', { context: 'Jetpack product name' } );
-	const socialAdvanced = translate( 'Social Advanced (Beta)', { context: 'Jetpack product name' } );
+	const socialBasic = translate( 'Social', { context: 'Jetpack product name' } );
+	const socialAdvanced = translate( 'Social', { context: 'Jetpack product name' } );
```
The display name must be Social on the pricing page.

CodeyGuyDylan commented on client/my-sites/plans/jetpack-plans/product-lightbox/constants.ts:
```diff
@@ -24,8 +26,14 @@ export const PRODUCT_OPTIONS: Record< string, TranslateResult > = {
+	} ),
 	[ PRODUCT_JETPACK_SOCIAL_ADVANCED ]: translate( 'Social Advanced', {
-		comment: 'Get unlimited shares',
+		comment: 'Get unlimited shares and share as a post.',
```
I find this text somewhat confusing. Does Social Basic not allow sharing as a post? Does sharing as a post mean adding an image/video?

spsiddarthan replied:
>Does Social Basic not allow sharing as a post? Does sharing as a post mean adding an image/video?

That's correct, we allow sharing as post only on the advanced plan.. I have updated the text to describe that better, let me know what you think!

CodeyGuyDylan commented on client/my-sites/plans/jetpack-plans/product-store/hooks/use-store-item-info.tsx:
```diff
@@ -256,6 +256,11 @@ export const useStoreItemInfo = ( {
+			const isOwned = getIsOwned( item );
+
+			if ( isOwned ) {
+				return <>{ translate( 'Manage Subscription' ) }</>;
```
I know this is the pattern in this function, I am just not sure the Fragments (<></>) are needed here? `translate` should just return a string in this scenario, maybe there's some reason I am unaware of?

spsiddarthan replied:
Good point, I will have that removed.

spsiddarthan replied:
```
if ( isProductInCart ) {
				return (
					<>
						<Gridicon icon="checkmark" />
						{ translate( 'View Cart' ) }
					</>
				);
			}
```

The same function had this bit which needs the `<React.Fragment>`, I reckon we added it to all. I have removed it where we don't need it.

CodeyGuyDylan reviewed (changes requested):
I tested this on an Atomic Ephemeral site and a Jurassic Ninja site, and for some reason, no matter which plan I buy, I can never get the pricing page to say "Manage Subscription"

This is on a site that has JETPACK_SOCIAL_BASIC_YEARLY
![image](https://user-images.githubusercontent.com/65001528/217612445-1a8b8f96-91e8-4fe1-a534-d5aea40f319b.png)
![image](https://user-images.githubusercontent.com/65001528/217612514-2a591b8a-c1b2-41f8-99a3-c0b0aae12f2b.png)

And if I press "Get", it takes me through the checkout process again and adds a year to my plan so that it is 2 years long

I am unsure if I am doing something wrong 🤔

spsiddarthan committed:
Address code review comments.

spsiddarthan commented:
>I can never get the pricing page to say "Manage Subscription"

I know we discussed this part on slack, just adding it here as well. :) We need to be logged on to the Jetpack cloud page to test the logged in version of the pricing page.

pablinos commented on client/my-sites/plans/jetpack-plans/product-store/hooks/use-store-item-info.tsx:
```diff
@@ -256,9 +256,14 @@ export const useStoreItemInfo = ( {
 		( item: SelectorProduct ) => {
 			const shouldShowCart = getShouldShowCart( item );
 			const isProductInCart = getIsProductInCart( item );
+			const isOwned = getIsOwned( item );
```
There is `productButtonLabel` used in the `getCtaLabel` function, that seems to handle this `isOwned` state etc. Should that be utilised for the lightbox too?

spsiddarthan replied:
This is resolved now!

pablinos reviewed (commented):
I'm not sure what we should have instead, but having the payment information here doesn't seem quite right to me, unless it's the payments I'm currently making. I imagine it would still say about the $1 trial even after the first month?

<img width="413" alt="image" src="https://user-images.githubusercontent.com/96462/217824738-4b94ddf5-e8aa-49d5-87c7-5b33e79b8f54.png">

I thought I discovered a problem with this change, but it's in production too. I've got a VaultPress subscription and so I clicked on 'Manage Subscription'. That caused it to be added to my cart, and then I received this message:

<img width="1508" alt="image" src="https://user-images.githubusercontent.com/96462/217827166-2860155c-ac79-4ae6-9ace-256099d85d6a.png">

I think we might want to update those Manage Subscription links too. It could be because I have the Security bundle, so it's not checking whether I have the product correctly.

spsiddarthan committed:
Use productButtonLabel to get the product lighthouse cta label as well.

spsiddarthan committed:
Merge branch 'trunk' of https://github.com/Automattic/wp-calypso into add/fix-social-product-name

spsiddarthan committed:
Show payment details only if the plan hasn't been purchased yet in the product lightbox.

spsiddarthan committed:
Added products and features.

spsiddarthan committed:
Added social products to the list of complete products.

spsiddarthan committed:
Remove unwanted change.

spsiddarthan committed:
Removed trial specific changes.

spsiddarthan committed:
Put back change.

spsiddarthan commented:
Thanks @pablinos for your review! I have addressed most of the issues. I also got rid of the monthly trial changes for the basic plan. 

> I'm not sure what we should have instead, but having the payment information here doesn't seem quite right to me, unless it's the payments I'm currently making. 

I have removed the payment information and from being displayed if you own the plan and just show a manage susbcription button. 

> I imagine it would still say about the $1 trial even after the first month

Yup, that information is displayed regardless of where the user is with the subscription. We should recommend to clean that up.

> I thought I discovered a problem with this change, but it's in production too. I've got a VaultPress subscription and so I clicked on 'Manage Subscription'. That caused it to be added to my cart, and then I received this message.

Good find, this happens when you have the complete or security plans. We should just take you to the manage susbcription page on click and not add the product to the cart. I discovered a few more issues when you own the security or the complete plan:
 - I noticed warnings fir duplicate keys in the console when you own the complete plan which I think is related to some products such as the CRM being displayed twice. 
 
 
 What I have done in this PR is show the manage susbcription button if you already have the complete plan, but clicking on manage subscription adds the product to the cart like you said in the case of backups and the security plan.
 
  I think we should fix the behaviour of what happens when you own a complete plan in a separate PR. Along with that, we should fix the aforementioned warning in the console and also fix duplicate products (CRM gets displayed twice if you have the complete plan) from being shown if you own the complete plan.

spsiddarthan committed:
Merge branch 'trunk' of https://github.com/Automattic/wp-calypso into add/fix-social-product-name

spsiddarthan committed:
Revert "Remove unwanted change."

This reverts commit 37b99460a9f46f909d35b8f9616e07f0b57528aa.

pablinos commented:
> I think we should fix the behaviour of what happens when you own a complete plan in a separate PR

Yes, that makes sense - keep this change about the Social plans.

> What I have done in this PR is show the manage susbcription button if you already have the complete plan,

Should that change be in a follow-up PR too? I suppose it depends on how quickly we follow-up, but it feels a bit broken having a 'Manage Subscription' button add something to your cart.

pablinos commented on packages/calypso-products/src/plans-list.tsx:
```diff
@@ -2311,7 +2318,7 @@ export const PLANS_LIST: Record< string, Plan | JetpackPlan | WPComPlan > = {
 			translate( 'Boost: Automatic CSS generation' ),
 			translate( 'Site Search: Up to 100k records and 100k requests/mo.' ),
-			translate( 'Social: Basic with unlimited shares' ),
+			translate( 'Social: Unlimited shares and share as a post.' ),
```
Is this the language we want to use? I think I saw elsewhere that we were going to describe it as 'share images and videos' or something.

pablinos reviewed (commented):
I've just discovered that this doesn't work the same in the Upgrades section of Calypso
<img width="919" alt="image" src="https://user-images.githubusercontent.com/96462/218461882-c1517c70-1d5a-4e1e-a14d-00488fd137c0.png">

There's no 'more about Social' link and the 'Manage Subscription' option doesn't open the modal. Is this expected? Should we change that?

spsiddarthan commented:
@pablinos, nice catch about that section on calypso - I didn't know it existed.  I will fix the issue you pointed out there.

>Should that change be in a follow-up PR too? I suppose it depends on how quickly we follow-up, but it feels a bit broken having a 'Manage Subscription' button add something to your cart.

Fair enough, I can put it in a different follow up PR.

spsiddarthan committed:
Remove changes introduced to include basic and advanced plans.

spsiddarthan committed:
Improve copy.

spsiddarthan commented:
>There's no 'more about Social' link  Is this expected? Should we change that?

@pablinos, 'more about <Product>' section gets hidden when you own the product. 

>the 'Manage Subscription' option doesn't open the modal. Is this expected?
On staging and prod, it's not fixed yet. This PR fixes it for both Calypso Blue and Green. I tested it by on Calypso Blue by booting it up on local and it functions correctly.

pablinos reviewed (approved):
> On staging and prod, it's not fixed yet. This PR fixes it for both Calypso Blue and Green. I tested it by on Calypso Blue by booting it up on local and it functions correctly.

Ah, I see. I got confused as it is working for the Cloud live link.

a8ci18n commented:
This Pull Request is now available for translation here: https://translate.wordpress.com/deliverables/7815430

Thank you @spsiddarthan for including a screenshot in the description! This is really helpful for our translators.

a8ci18n commented:
Translation for this Pull Request [has now been finished](https://translate.wordpress.com/deliverables/overview/7815430).
